plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/BaseTreeNode.java

```diff
@@ -58,16 +60,34 @@ public void removeChildren() {
 	public void setImageDescriptor(ImageDescriptor imageDescriptor) {
 		this.imageDescriptor = imageDescriptor;
 	}
-	
+
 	protected ImageDescriptor getImageDescriptor(IResource object) {
-		ILabelProvider labelProvider = WorkbenchLabelProvider.getDecoratingWorkbenchLabelProvider();
+		ILabelProvider labelProvider = null;
+		Image image = null;
+
 		try {
-			Image image = labelProvider.getImage(object);
-			if (image == null || image.isDisposed())
+			labelProvider = WorkbenchLabelProvider.getDecoratingWorkbenchLabelProvider();
+			image = labelProvider.getImage(object);
+
+			if (image == null || image.isDisposed()) {
 				return null;
-			
+			}
+
 			return getImageDescriptorFromImage(image);
+
+		} catch (Exception e) {
+			SnykLogger.logError(e);
+			return null;
 		} finally {
+			disposeResources(labelProvider, image);
+		}
+	}
+
+	private void disposeResources(ILabelProvider labelProvider, Image image) {
+		if (image != null && !image.isDisposed()) {
+			image.dispose();
+		}
+		if (labelProvider != null) {
```

Comment:
I don't think that this is enough. Most likely, we'll also need to cache imageDescriptors for commonly used pics (e.g. severity icons) and renew them once they are disposed.